internal/controller/apiextensions/claim/configurator.go

```diff
@@ -136,13 +128,8 @@ func (c *APIDryRunCompositeConfigurator) Configure(ctx context.Context, cm resou
 	ucp.SetClaimReference(proposed)
 
 	if !meta.WasCreated(cp) {
-		// The API server returns an available name derived from
-		// generateName when we perform a dry-run create. This name is
-		// likely (but not guaranteed) to be available when we create
-		// the composite resource. If the API server generates a name
-		// that is unavailable it will return a 500 ServerTimeout error.
-		cp.SetGenerateName(fmt.Sprintf("%s-", cm.GetName()))
-		return errors.Wrap(c.client.Create(ctx, cp, client.DryRunAll), errName)
+		// XR name follows the convention <claim_name>-<claim_uid_suffix>
+		cp.SetName(fmt.Sprintf("%s-%s", cm.GetName(), cm.GetUID()[19:]))
```

Comment:
curious, why `19:`? Why not `:8` or so?

Reply:
what happens if the partial UID conflicts? Can it realistically? What's the propability?

Reply:
wondering whether the format of the uid is specified in kube. Probably not. Could be different tomorrow such that `19:` breaks.

Reply:
> curious, why 19:? Why not :8 or so?

UUIDs are 128 bits long, so I wanted to take half them (64) to have enough entropy, i.e. to avoid collisions. Ofcourse, we can take more bits (or less), but that would then have impact on collision and on max length of claim name.

> wondering whether the format of the uid is specified in kube. Probably not. Could be different tomorrow such that 19: breaks.

Kubernetes doc: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#uids and more https://en.wikipedia.org/wiki/Universally_unique_identifier#Encoding

It is a well know standard. Not sure how realistic is that kubernetes switches to something else. They could switch to different UUID encoding, but if we want to guard against that, we could [parse the string representation](https://pkg.go.dev/github.com/google/uuid#Parse) and then encode again the suffix we would like to take according to some schema.

> what happens if the partial UID conflicts? Can it realistically? What's the propability?

Kuberenets uses Version 4 UUID, which are randomly generated, and the probability for collision is very low:

https://en.wikipedia.org/wiki/Universally_unique_identifier#Collisions

Even if we take 64 bits, it is reasonably low that I think we should not add here any logic that should try to handle it.

Reply:
just put that into the comment. It also matters here that conflict only happens for equal claim names (cross namespaces). I.e. with 10000ish namespaces and equal claim names:

<img width="1019" alt="Bildschirmfoto 2023-10-06 um 16 58 03" src="https://github.com/crossplane/crossplane/assets/730123/303b15f1-4c83-4a79-81f5-3314f5ab78f8">

2^-12 should be safe.

Reply:
@sttts I have made the code agnostic to the encoding format used by the k8s.